src/libmongoc/src/mongoc/mongoc-client-pool.c

```diff
@@ -357,6 +363,76 @@ mongoc_client_pool_try_pop (mongoc_client_pool_t *pool)
    RETURN (client);
 }
 
+typedef struct {
+   mongoc_array_t *server_ids;
+   mongoc_cluster_t *cluster;
+} prune_ctx;
+
+static int
+server_id_cmp (const void *a_, const void *b_)
+{
+   const uint32_t *const a = (const uint32_t *) a_;
+   const uint32_t *const b = (const uint32_t *) b_;
+
+   if (*a == *b) {
+      return 0;
+   }
+
+   return a < b ? -1 : 1;
```

Comment:
Confirming, is `a < b` intentional instead of `*a < *b`? Recommend adding a comment here documenting so due to its unusual behavior.

Reply:
No, that was a (very well spotted) bug. Fixed.